apps/src/sites/studio/pages/foorm/forms/editorHelpers.js

```diff
@@ -8,6 +11,29 @@ import _ from 'lodash';
 
 let codeMirror;
 
+const nameKeyRegex = new RegExp(/"(?:name|value)"\:\s*"(.+)",?/gi);
```

Comment:
This is (classically regex) gibberish, could you add a comment describing what these are supposed to match?

Reply:
Relatedly, is there a reason not to just parse the text as JSON, then figure out what keys are problematic? I pulled the branch to understand how the errors will appear (looks great BTW), but I'm seeing some errors appear in places I wouldn't expect:

![image](https://user-images.githubusercontent.com/25372625/118154505-fb40e800-b3cb-11eb-851f-ab2a1328cc72.png)

Reply:
the problem is that to integrate with the codemirror error display, you need the line/character numbers, which would be difficult to get back from the json representation. but yes on comments

Reply:
i had the validator check "name" and "value" keys because the values also get turned into keys in the case of matrix questions

Reply:
Ah right, good call on matrix questions!

I was trying to come up with an error message that clarifies what exactly can only contain letters and underscores (I think "question keys" might not mean much to end users), but I'm not coming up with a great message. I think fine to leave as-is unless you have an idea on how to clarify.